internal/ignition/ignition.go

```diff
@@ -1394,7 +1410,22 @@ func (ib *ignitionBuilder) FormatSecondDayWorkerIgnitionFile(address string, mac
 	if err = tmpl.Execute(buf, ignitionParams); err != nil {
 		return nil, err
 	}
-	return buf.Bytes(), nil
+
+	overrides := buf.String()
+	if host.IgnitionConfigOverrides != "" {
+		overrides, err = MergeIgnitionConfig(buf.Bytes(), []byte(host.IgnitionConfigOverrides))
+		if err != nil {
+			return []byte(""), errors.Wrapf(err, "Failed to apply ignition override for host %s", host.ID)
+		}
+		ib.log.Infof("Applied ignition override for host %s", host.ID)
+	}
+
+	res, err := SetHostnameForNodeIgnition([]byte(overrides), host)
```

Comment:
Why do we need this?
I think we want to do it only if the user requested to set the hostname via assisted.

Reply:
I did not create that myself, it's only moving this code around. Originally it lived in

```
func (b *bareMetalInventory) createAndUploadNodeIgnition
```

which has a wrong name because it applies only to the day2-node